OnlineShop/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using DataAccess.EF;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using DataAccess.EF.Models;
+
+namespace OnlineShop.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoriesController : ControllerBase
+    {
+        public CategoriesController(IUnitOfWork uow)
+        {
+            _uow = uow;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<Category>> GetCategories()
+        {
+            return _uow.CategoryRepository.Get().ToList();
+        }
+
+        [HttpGet("{id:int}")]
+        public ActionResult<Category> GetCategory(int id)
+        {
+            var category = _uow.CategoryRepository.Find(id);
+
+            if (category == null)
+            {
+                return NotFound();
+            }
+
+            return category;
+        }
+
+        [HttpPut("{id:int}")]
+        public IActionResult PutCategory(int id, Category? category)
+        {
+            if (category == null || id != category.Id) return BadRequest();
+
+            try
+            {
+                _uow.CategoryRepository.Update(category);
+                _uow.Save();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (_uow.CategoryRepository.Find(id) == null) return NotFound();
+                throw;
+            }
+
+            return NoContent(); ;
+        }
+
+        [HttpPost]
+        public ActionResult<Category> PostCategory(Category? category)
```

Comment:
remove ?, and you will have the same behavior. Bundings will do all wark for you. Add !! and you will punish consumer